### ISSUE TITLE
Publish Debian based Docker image

### ISSUE THREAD

**Issue by carlsmedstad** [koalaman/shellcheck#2327]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

Is there opinions against building and publishing a Debian based docker image? While being a lot larger than Alpine (70 MB vs. 6 MB), it would enable the usage of standard GNU tools.

The use-case I'm currently thinking of is a CI job that recursively finds all shell files and lints them:
```sh
grep \
  --recursive \
  --exclude-dir .git \
  --extended-regexp \
  --files-with-matches \
  '^#!/bin/(bash|sh|env (bash|sh))' \
  . \
  | xargs shellcheck
```
Long-form options are not allowed and the option `--exclude-dir` is not available at all on Alpine's grep, unfortunately.

There is of course the possibility for us to build our own image, but this seems like a not too uncommon use-case and a `shellcheck-debian` image would serve as a good option to `shellcheck-alpine`.

If no objections to this suggestion I can make a PR.

**Comment by koalaman** [comment]
I'd really prefer not maintaining more Docker images upstream. Consider one of the following instead:

* Rewriting this for POSIX and running it on `shellcheck-alpine`, e.g. `find . -name .git -prune -o -type f -exec grep -lE '^#!/bin/(bash|sh|env (bash|sh))' {} +`
* Installing GNU grep with `apk add grep`
* If the CI allows it, doing `grep` outside Docker and using `xargs docker run ...` instead
* Including ShellCheck in the distro of your choice via the Dockerfile:

```
FROM koalaman/shellcheck:v0.7.2 AS shellcheck
FROM debian:latest
COPY --from=shellcheck /bin/shellcheck /bin/
```

**Comment by carlsmedstad** [comment]
Those are great suggestions, thanks!
